core/domain/suggestion_services.py

```diff
@@ -588,13 +588,25 @@ def get_all_suggestions_that_can_be_reviewed_by_user(user_id):
     ])
 
 
-def get_reviewable_suggestions(user_id, suggestion_type):
+def get_reviewable_suggestions(user_id, suggestion_type,
+                               opportunity_summary_exp_ids, topic_name):
     """Returns a list of suggestions of given suggestion_type which the user
-    can review.
+    can review. If suggestion_type is translate_content, we only fetch
+    those suggestions which match with the passed opportunity ids.
```

Comment:
Let's try not to have args in a function which depend on each other -- in general, it's easier to maintain the code if each arg can independently take any value that is valid.

I think I would suggest making a new get_reviewable_translation_suggestions(user_id, opportunity_summary_exp_ids=None) function. Have this original function call that other function for the translations case, and have your new controller call the new function. That should be cleaner.

Reply:
Sorry, I didn't get you. Just wanted to ask, which arg are you pointing towards that is taking invalid value and are dependent on each other? Are you hinting towards 'None' as an invalid value?

Reply:
Sorry for being unclear. I mean that opportunity_summary_exp_ids seems to depend on suggestion_type being 'translate', and the case where it gets passed when the suggestion_type is not 'translate' seems to have undefined behaviour.

Reply:
Oh, so @sagangwee and I discussed and decided that since this feature is a pretty urgent request, we will only add topic filtering logic for suggestion_type=translate at the moment. But later on, we will extend it to suggestion_type=question as well. 

Once that is done, we will not have the dependent behaviour. Maybe we can go ahead and add a TODO then?

Reply:
I think my suggestion above (in the first comment in this thread) would still work? Let's not introduce tech debt unnecessarily, we can refactor again later when the new functionality comes in but let's keep the code clean at all times.

Reply:
Yes, I agree, the suggestion is still valid, but I was just making sure the reasoning of doing this re-factoring is clear.  Just to confirm and for my understanding, the reason being, design of the function's API is complicated.

Reply:
Yup that's right.

Reply:
This is complete.